rustvmm_gen/lib_kernel_source.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025 © Institute of Software, CAS. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import re
+import tarfile
+import requests
+import subprocess
+import shutil
+from urllib.parse import urlparse
+
+KERNEL_ORG_CDN = "https://cdn.kernel.org/pub/linux/kernel"
+
+
+def check_kernel_version(version):
+    """
+    Validate if the input kernel version exists in remote. Supports both X.Y
+    (namely X.Y.0 and .0 should be omitted) and X.Y.Z formats
+    """
+    # Validate version format
+    if not re.match(r"^\d+\.\d+(\.\d+)?$", version):
+        return (False, "Invalid version format. Use X.Y or X.Y.Z")
+
+    main_ver = version.split(".")[0]
+    base_url = f"{KERNEL_ORG_CDN}/v{main_ver}.x/"
+    tarball = f"linux-{version}.tar.xz"
+
+    try:
+        # Fetch content of `base_url`
+        response = requests.get(base_url, timeout=15)
+        response.raise_for_status()
+
+        # Check for exact filename match
+        if tarball in response.text:
+            print(f"Kernel version {version} found in remote")
+            return (True, "")
+
+        return (False, f"Kernel version {version} not found in remote")
+
+    except requests.exceptions.HTTPError as e:
+        if e.response.status_code == 404:
+            return (False, f"Kernel series v{main_ver}.x does not exist")
+        return (False, f"HTTP error ({e.response.status_code}): {str(e)}")
+    except requests.exceptions.Timeout:
+        return (False, "Connection timeout while checking version")
+    except requests.exceptions.RequestException as e:
+        return (False, f"Network error: {str(e)}")
+
+
+def create_temp_dir(version):
+    temp_dir = os.path.join("/tmp", f"linux-{version}-prepare")
+
+    if os.path.exists(temp_dir):
+        shutil.rmtree(temp_dir)
+
+    try:
+        os.makedirs(temp_dir)
+        return temp_dir
+    except OSError as e:
+        raise RuntimeError(f"Failed to create temp directory: {e}") from e
```

Comment:
Let's use something that relies on `mkdtemp(2)` here to avoid race conditions and the "delete if exists" dance: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory